packages/editor/src/utils/with-history/index.js

```diff
@@ -120,9 +114,13 @@ const withHistory = ( options = {} ) => ( reducer ) => {
 		let lastActionToSubmit = previousAction;
 
 		if (
-			shouldCreateUndoLevel ||
-			! past.length ||
-			! shouldOverwriteState( action, previousAction )
+			// Ignored action should never create an undo level, regardless of
+			// whether there is already history or not.
```

Comment:
What does this change, functionally? Does it fix something?

Reply:
Yes. If there is a change in selection, it should not create an undo level. Without this change, it would. I'm not sure why it was made to behave like that. Actions to ignore should always be ignored, regardless of whether there is already anything recorded or not.

Reply:
> If there is a change in selection, it should not create an undo level.

Yeah, in this case where selection is a part of history it makes a lot of sense.

> I'm not sure why it was made to behave like that.

I'd try to find something in the origins of `withHistory`; maybe there's a good reason, maybe it no longer applies. AFAIK that was mostly Andrew's work, so we can't just ask him. :)

Reply:
@mcsf I actually created a lot of this. :) It was introduced in #4956, but I forgot why I included this condition. I probably didn't think further than the use cases at the time. In addition all tests still pass, which should be a good sign.

Reply:
> @mcsf I actually created a lot of this. :) It was introduced in #4956

Oh, my bad. 😓 I guess I equated the `withHistory` higher-order reducer with the original history work.

> In addition all tests still pass, which should be a good sign.

Yeah, I should've mentioned this earlier too. :)